### PR TITLE
Remove trivial propertys. Rename some files.

### PR DESCRIPTION
**Pull request by Xazax-hun**
Trivial getter/setter propertys do not add any value to the code, only decrease the readability and increase code size. Once we need to do something special, we can add them back as we need them.

Also get rid of the generic package terminology in the file names. We no longer have multiple packages.

### CODE REVIEW

File: libcodechecker/server/database/database.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def __enter__(self):
         except Exception as ex:
             LOG.debug("Connection error")
             LOG.debug(ex)
-            self.error = DBStatus.FAILED_TO_CONNECT
+            self.db_error = DBStatus.FAILED_TO_CONNECT
```
Comment (gyorb) [inline libcodechecker/server/database/database.py]: `DBContext.error` is checked at multiple places, if we rename it we should rename it everywhere.

File: libcodechecker/analyze/analyzers/analyzer_clangsa.py [hunk 2]
```diff
@@ -104,8 +104,7 @@ def add_checker_config(self, checker_cfg):
 
         self.__checker_configs.append(checker_cfg)
 
-    @staticmethod
-    def get_analyzer_checkers(config_handler, env):
+    def get_analyzer_checkers(self, config_handler, env):
```
[the remark this answers is not in the document]
Comment (Xazax-hun) [reply libcodechecker/analyze/analyzers/analyzer_clangsa.py]: There is a method with the same signature in the base class. I added the self to make this function a proper override.

File: libcodechecker/analyze/statistics_collector.py [hunk 3]
```diff
@@ -29,14 +29,9 @@ def build_stat_coll_cmd(action, config, source, environ):
...
-    cmd.append('-c')
-    cmd.extend(['-x', action.lang])
-
-    # Do not warn about the unused gcc/g++ arguments.
-    cmd.append('-Qunused-arguments')
-
-    cmd.append('--analyze')
+    cmd = [config.analyzer_binary, '-c', '-x', action.lang, '--analyze',
+           # Do not warn about the unused gcc/g++ arguments.
+           '-Qunused-arguments']
```
[the remark this answers is not in the document]
Comment (Xazax-hun) [reply libcodechecker/analyze/statistics_collector.py]: I think it should not matter. The order only matters for positional arguments, command line switches with arguments (e.g -I) and if we are setting the same thing multiple times (like optimization level).

File: libcodechecker/server/database/database.py (hunk 1 above)
Comment (gyorb) [reply libcodechecker/server/database/database.py]: Ok, thanks. Please initialize `self.db_error` in `__intit__`

**Comment by Xazax-hun** [comment]
Done.

**Pull request merged**